Services/source/updater/updater.c

```diff
@@ -73,49 +87,101 @@ SAT_returnState updater_app(csp_packet_t *packet) {
                   status = -1; break;
               }
               flash_destination += flash_size;
-          }
-          status = 0;
-          break;
 
-      case GET_GOLDEN_INFO:
-          if (!init_eeprom()) {
-              status = -1; break;
-          }
-          app_info = priv_eeprom_get_golden_info();
-          status = 0;
-          memcpy(&packet->data[OUT_DATA_BYTE], &app_info,sizeof(app_info));
-          set_packet_length(packet, sizeof(int8_t)+sizeof(app_info) + 1);
-          break;
-
-      case GET_APP_INFO:
-          if (!init_eeprom()) {
-              status = -1; break;
-          }
-          app_info = priv_eeprom_get_app_info();
-          status = 0;
-          memcpy(&packet->data[OUT_DATA_BYTE], &app_info,sizeof(app_info));
-          set_packet_length(packet, sizeof(int8_t)+sizeof(app_info) + 1);
-          break;
-
-
-      case SET_GOLDEN_INFO:
-          memcpy(&packet->data[IN_DATA_BYTE], &app_info, sizeof(app_info));
-          priv_eeprom_set_golden_info(app_info);
-          set_packet_length(packet, sizeof(int8_t) + 1);
-          status = 0;
-          break;
-
-      case SET_APP_INFO:
-          memcpy(&packet->data[IN_DATA_BYTE], &app_info, sizeof(app_info));
-          priv_eeprom_set_app_info(app_info);
-          set_packet_length(packet, sizeof(int8_t) + 1);
-          status = 0;
-          break;
-
-      default:
-        ex2_log("No such subservice\n");
-        shutdown_eeprom();
-        return SATR_PKT_ILLEGAL_SUBSERVICE;
+              status = 0;
+    #else
+              ex2_log("FAILED Attempt to flash from non golden image");
+              status = -1;
+    #endif
+              break;
+
+          case GET_GOLDEN_INFO:
+
+              image_info golden_info = priv_eeprom_get_golden_info();
+              status = 0;
+              memcpy(&packet->data[OUT_DATA_BYTE], &app_info,sizeof(golden_info));
```

Comment:
We should also make sure to check that the CSP packet has enough space before every memcpy using `size_t csp_buffer_data_size(void) ` : 

```
if (sizeof(golden_info) + 1 <= csp_buffer_data_size()) {
  // do memcpy
} else // don't do memcpy, or truncate it, or whatever makes sense.
```
Maybe a wrapper around memcpy would be helpful to not have to write this every time?

Reply:
In fact, maybe something that would be useful to you would be a memcpy wrapper that also updates the packet size -- something like this could live in `service.c`. If you don't want to do this for this PR could you make an issue in git to do this?

Reply:
I'll make it an issue. Every memcpy in ex2_services would have to be updated to make it worth it

Reply:
Sounds good. Might be a good 'first issue'